### PR TITLE
aruha-1375 fix concurrency issue on timelines switch

### PR DESCRIPTION
**Pull request by rcillo**
## Description

When switching timelines, under high load with multiple concurrent
producers, events would be published to the old timeline, making them
unaccessible to consumers.

The problem was caused by a concurrency issue, since cache expiration
and event publishing would be accidentally executed in parallel.

In this PR we first expire the cache to only then allow producers to
proceed with publishing.
## Review
- [x] CHANGELOG

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/service/timeline/LocalLocking.java [hunk 1]
```diff
@@ -70,4 +45,34 @@ public Closeable workWithEventType(final String eventType, final long timeoutMs)
...
+        final Set<String> unlockedEventTypes = new HashSet<>();
+        synchronized (lock) {
+            for (final String item : this.lockedEventTypes) {
+                if (!lockedEventTypes.contains(item)) {
+                    unlockedEventTypes.add(item);
+                }
+            }
+        }
+        return unlockedEventTypes;
+    }
```
Comment (antban) [inline src/main/java/org/zalando/nakadi/service/timeline/LocalLocking.java]: Isn't it simplier 
```
synchronized (lock) {
    return this.lockedEventTypes.stream()
            .filter(v -> !lockedEventTypes.contains(v))
            .collect(Collectors.toSet));
}
```

**Comment by antban** [comment]
:+1:

File: src/main/java/org/zalando/nakadi/service/timeline/LocalLocking.java (hunk 1 above)
Comment (rcillo) [reply src/main/java/org/zalando/nakadi/service/timeline/LocalLocking.java]: 👍 done here https://github.com/zalando/nakadi/pull/796/commits/0c2f80fa08a4b8087474b97673f34fa13ffef7b3

**Comment by antban** [comment]
:+1:

**Comment by rcillo** [comment]
👍

**Comment by rcillo** [comment]
deploy validation please

**Comment by rcillo** [comment]
👍

**Comment by antban** [comment]
:+1:

**Pull request merged**